core/src/main/java/hudson/tools/ZipExtractionInstaller.java

```diff
@@ -100,7 +100,15 @@ public FormValidation doCheckUrl(@QueryParameter String value) {
                 URLConnection conn = ProxyConfiguration.open(new URL(value));
                 conn.connect();
                 if (conn instanceof HttpURLConnection) {
-                    if (((HttpURLConnection) conn).getResponseCode() != HttpURLConnection.HTTP_OK) {
+                    // A redirection from http to https (or vise versa) returns a 302 response status. Force redirection
+                    int responseCode = ((HttpURLConnection)conn).getResponseCode();
+                    if (HttpURLConnection.HTTP_MOVED_PERM == responseCode || HttpURLConnection.HTTP_MOVED_TEMP == responseCode || HttpURLConnection.HTTP_SEE_OTHER == responseCode) {
+                        // In case of redirection, we have to connect to the new URL
+                        String redirection = ((HttpURLConnection) conn).getHeaderField("Location");
```

Comment:
Why is this inline instead of using the method in `ProxyConfiguration`?

Reply:
In previous code of `ZipExtractionInstaller` class the message shown was different between the one if IOException is thrown and the one in case of a non OK status. Using the new created method we loose this behaviour and I didn't want to develop such an intrusive solution. But if it is OK to use the same validation message in both cases, I fully agree about using this new method.

Reply:
Changed